internal/common/voting/VotingEngine.go

```diff
@@ -0,0 +1,72 @@
+package voting
+
+import (
+	"github.com/google/uuid"
+)
+
+// Generic IVoter type to accept different outputs
+type IVoter interface {
+	GetVotes() map[uuid.UUID]float64
+}
+
+// lootboxID:distribution
+type LootboxVoteMap map[uuid.UUID]float64
+
+// LootboxVoteMap already has the required structure, so we just add a method to satisfy the IVoter interface.
+func (lvm LootboxVoteMap) GetVotes() map[uuid.UUID]float64 {
+	return lvm
+}
+
+// BikerID:distribution
+type IdVoteMap map[uuid.UUID]float64
+
+func (ivm IdVoteMap) GetVotes() map[uuid.UUID]float64 {
+	return ivm
+}
+
+// this function will take in a list of maps from ids to their corresponding vote (yes/ no in the case of acceptance)
+// and retunr a list of ids that can be accepted according to some metric (ie more than half voted yes)
+// ranked according to a metric (ie overall number of yes's)
+func GetAcceptanceRanking([]map[uuid.UUID]bool) []uuid.UUID {
+	// TODO implement
+	panic("not implemented")
+	// return make([]uuid.UUID, 0)
+}
+
+// Returns the normalized vote outcome (assumes all the maps contain a voting between 0-1
+// for each option, and that all the votings sum to 1)
```

Comment:
based on this assumption, you'd want to throw an error if this condition isn't met

Reply:
i.e.,

```
if !sum(voteMap) == 1.0 {
throw new error(...)
}
```